app/org/sagebionetworks/bridge/models/subpopulations/SubpopulationGuidImpl.java

```diff
@@ -0,0 +1,36 @@
+package org.sagebionetworks.bridge.models.subpopulations;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Objects;
+
+public final class SubpopulationGuidImpl implements SubpopulationGuid {
```

Comment:
As per IM conversation, we ran into a lot of trouble with this for both Studies and Surveys, specifically when we passed in the mutable full object when we actually needed an immutable ID holder, or when we unnecessarily loaded the full object when we only needed.

What we should do instead is have Subpop and SubpopGuid be separate objects, so you can't mistakenly use a Subpop for a SubpopGuid.